### ISSUE TITLE
Splash Screen doesn't go away on Android

### ISSUE THREAD

**Issue by AugustoAleGonZhipCode** [crazycodeboy/react-native-splash-screen#31]
Hello.

I am just starting my Android application and I was testing this plugin. I follow every single step that it is in the instructions and it works but the splash screen doesn't go away, stay fixed.

Can you give me some help in order to make it done?

**Comment by tilarm** [comment]
Same here

**Comment by joaohkfaria** [comment]
Yes, it's not working here too. One way to bypass this problem on Android is to add on your onCreate method in MainAcitivity this code:

```java
final Activity activity = this;
final Handler handler = new Handler();
handler.postDelayed(new Runnable() {
    @Override
    public void run() {
        SplashScreen.hide(activity);
    }
}, 3000);
```

This will hide the SplashScreen after 3000 ms (you can change to the value that you want.

**Comment by AugustoAleGonZhipCode** [comment]
Yes. One of the way of doing it is the Android approach. Thanks for your support. @joaohkfaria 
Nevertheless I would like to contribute to the Android platform.

**Comment by brunocascio** [comment]
@joaohkfaria 

`final Activity activity = this;` is not possible 😢 . Currently, `this` is an instance of Application, not Activity.

**Comment by MeganGilligan1** [comment]
Hi everyone, I had this issue but realised I'd forgotten to do one of the steps:


 import SplashScreen from 'react-native-splash-screen'

export default class WelcomePage extends Component {

    componentDidMount() {
         // do anything while splash screen keeps, use await to wait for an async task.
        **SplashScreen.hide();**
    }
}

Hope this helps

**Comment by MossP** [comment]
it seems @MeganGilligan1 has pointed out the issue here @AugustoAleGonZhipCode. I think this should now be closed.

**Comment by camux** [comment]
I dont think must be closed, because @MeganGilligan1 solution no resolved root of problem, Indeed MeganGilligan1 solution show error to me.

/android/app/src/main/java/com/melhous/MainActivity.java:27: error: <identifier> expected
    handler.postDelayed(new Runnable() {
                       ^

**Comment by joaomilho** [comment]
So I have a similar issue, but in my case the app itself works fine but the splash screen is always there in storybooks. The browser displays things correctly, but the emulator is always in the splash state. Any idea?
